src/Event/NamedType.php

```diff
@@ -0,0 +1,30 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Event;
+
+final class NamedType implements Type
+{
+    private $name;
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    public function is(Type $other): bool
+    {
+        return $other->asString() === $this->asString();
```

Comment:
This seems prone to collision with other types.
Maybe change to `get_class($other) === get_class($this) && $other->asString() === $this->asString()`?

Note that in PHP the object has access to the `$name` property of `$other` since it's in the same class, so this would also work:

```
return get_class($other) === get_class($this) && $other->name === $this->name;
```

Reply:
As this class will vanish (see below), this point is probably mute.

Regarding property access: Yes, we are aware of that but I consider it bad software design. The `asString` is part of the public API, the existence of the property `name` is a (private) implementation detail.